Sources/SwiftRefactor/ConvertComputedPropertyToZeroParameterFunction.swift

```diff
@@ -0,0 +1,77 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#if swift(>=6)
+public import SwiftSyntax
+#else
+import SwiftSyntax
+#endif
+
+public struct ConvertComputedPropertyToZeroParameterFunction: SyntaxRefactoringProvider {
+  public static func refactor(syntax: VariableDeclSyntax, in context: Void) -> FunctionDeclSyntax? {
+    guard syntax.bindings.count == 1,
+      let binding = syntax.bindings.first,
+      var identifierPattern = binding.pattern.as(IdentifierPatternSyntax.self)
+    else { return nil }
+
+    identifierPattern.leadingTrivia = syntax.bindingSpecifier.trailingTrivia
+
+    guard let typeAnnotation = binding.typeAnnotation,
+      let accessorBlock = binding.accessorBlock,
+      case let .getter(statements) = accessorBlock.accessors
+    else { return nil }
+
+    let returnType = typeAnnotation.type
+
+    var returnClause: ReturnClauseSyntax?
+    var trailingTrivia: Trivia = []
+
+    if !returnType.isVoid {
+      let trailingTrivia = typeAnnotation.colon.leadingTrivia + identifierPattern.identifier.trailingTrivia
+      returnClause = ReturnClauseSyntax(
+        arrow: .arrowToken(
+          leadingTrivia: trailingTrivia.isEmpty ? .space : trailingTrivia,
+          trailingTrivia: typeAnnotation.colon.trailingTrivia
+        ),
+        type: returnType
+      )
+    } else {
+      trailingTrivia = typeAnnotation.colon.trailingTrivia
+    }
+
+    let body = CodeBlockSyntax(
+      leftBrace: accessorBlock.leftBrace,
+      statements: statements,
+      rightBrace: accessorBlock.rightBrace
+    )
+
+    let functionSignature = FunctionSignatureSyntax(
+      parameterClause: FunctionParameterClauseSyntax(parameters: []),
+      returnClause: returnClause,
+      trailingTrivia: trailingTrivia
+    )
+
+    return FunctionDeclSyntax(
+      modifiers: syntax.modifiers,
+      name: identifierPattern.identifier.with(\.trailingTrivia, []),
+      signature: functionSignature,
+      body: body
+    )
+  }
+}
+
+fileprivate extension TypeSyntax {
+  var isVoid: Bool {
+    let description = trimmedDescription
+    return description == "Void" || description == "()"
+  }
+}
```

Comment:
We could inspect the syntax tree more syntactically. That way, we also match eg. `( )` with a space between the parentheses (not that I expect anyone to write it like that but still).

```suggestion
fileprivate extension TypeSyntax {
  var isVoid: Bool {
    switch self.as(TypeSyntaxEnum.self) {
    case .identifierType(let identifierType) where identifierType.name.text == "Void": return true
    case .tupleType(let tupleType) where tupleType.elements.isEmpty: return true
    default: return false
    }
  }
}
```

Reply:
Wow, this is the better way to handle it 👍🏻